android/beagle/src/main/java/br/com/zup/beagle/android/components/ListViewTwo.kt

```
@@ -172,25 +197,39 @@ internal class ListViewContextAdapter2(
-        this@ListViewContextAdapter2.rootView.generateViewModelInstance<ScreenContextViewModel>().linkBindingToContextAndEvaluateThem()
+    override fun onBindViewHolder(holder: ContextViewHolderTwo, position: Int) {
+        holder.itemView.setContextData(ContextData(id = "item", value = adapterItens[position].data))
+
+        val innerRecycler = findNestedRecyclerView(holder.itemView)
+        innerRecycler?.let{
+            if(!adapterItens[position].initialized) {
+                val recyclerAdapter = it.adapter as ListViewContextAdapter2
+                adapterItens[position].childAdapter =
+                    ListViewContextAdapter2(
+                        recyclerAdapter.template,
+                        recyclerAdapter.viewFactory,
+                        recyclerAdapter.orientation,
+                        recyclerAdapter.rootView,
+                        recyclerAdapter.onInit
+                    )
+                adapterItens[position].initialized = true
+                adapterItens[position].childAdapter?.executeInit(it)
+            }
+
+            it.adapter = adapterItens[position].childAdapter
+            it.adapter?.notifyDataSetChanged()
+        }
     }
 
-    override fun onBindViewHolder(holder: ContextViewHolderTwo, position: Int) {
-        val item = listItems[position]
-        val view = holder.itemView
-        view.setContextData(ContextData(id = "item", value = item))
-        view.getContextBinding()?.let { contextBinding ->
-            rootView.generateViewModelInstance<ScreenContextViewModel>().notifyBindingChanges(contextBinding)
+    fun executeInit(recyclerView: RecyclerView){
+        onInit?.forEach { action ->
+            action.execute(rootView, recyclerView)
         }
     }
 
     fun setList(list: List<Any>) {
         try {
-            listItems = ArrayList(list)
+            adapterItens = list.map { BeagleAdapterItem(it) }
             notifyDataSetChanged()
         } catch (e: Exception) {
```

<table data-meta="generated_by_android">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/f9e7d22b96931838e54e85b390cb6318fe3ad197/android/beagle/src/main/java/br/com/zup/beagle/android/components/ListViewTwo.kt#L234"></span>Empty catch block detected. If the exception can be safely ignored, name the exception according to one of the exemptions as per the configuration of this rule.</td>
    </tr>
  </tbody>
</table>